gtsam/discrete/tests/testDiscreteBayesTree.cpp

```diff
@@ -26,76 +26,89 @@ using namespace boost::assign;
+   * and then create the Bayes tree from it.
+   */
+  TestFixture() {
+    // Define variables.
+    for (int i = 0; i < 15; i++) {
+      DiscreteKey key_i(i, 2);
+      keys.push_back(key_i);
+    }
 
-TEST_UNSAFE(DiscreteBayesTree, ThinTree) {
-  const int nrNodes = 15;
-  const size_t nrStates = 2;
+    // Create thin-tree Bayesnet.
+    bayesNet.add(keys[14] % "1/3");
 
-  // define variables
-  vector<DiscreteKey> key;
-  for (int i = 0; i < nrNodes; i++) {
-    DiscreteKey key_i(i, nrStates);
-    key.push_back(key_i);
-  }
+    bayesNet.add(keys[13] | keys[14] = "1/3 3/1");
+    bayesNet.add(keys[12] | keys[14] = "3/1 3/1");
 
-  // create a thin-tree Bayesnet, a la Jean-Guillaume
-  DiscreteBayesNet bayesNet;
-  bayesNet.add(key[14] % "1/3");
+    bayesNet.add((keys[11] | keys[13], keys[14]) = "1/4 2/3 3/2 4/1");
+    bayesNet.add((keys[10] | keys[13], keys[14]) = "1/4 3/2 2/3 4/1");
+    bayesNet.add((keys[9] | keys[12], keys[14]) = "4/1 2/3 F 1/4");
+    bayesNet.add((keys[8] | keys[12], keys[14]) = "T 1/4 3/2 4/1");
 
-  bayesNet.add(key[13] | key[14] = "1/3 3/1");
-  bayesNet.add(key[12] | key[14] = "3/1 3/1");
+    bayesNet.add((keys[7] | keys[11], keys[13]) = "1/4 2/3 3/2 4/1");
+    bayesNet.add((keys[6] | keys[11], keys[13]) = "1/4 3/2 2/3 4/1");
+    bayesNet.add((keys[5] | keys[10], keys[13]) = "4/1 2/3 3/2 1/4");
+    bayesNet.add((keys[4] | keys[10], keys[13]) = "2/3 1/4 3/2 4/1");
 
-  bayesNet.add((key[11] | key[13], key[14]) = "1/4 2/3 3/2 4/1");
-  bayesNet.add((key[10] | key[13], key[14]) = "1/4 3/2 2/3 4/1");
-  bayesNet.add((key[9] | key[12], key[14]) = "4/1 2/3 F 1/4");
-  bayesNet.add((key[8] | key[12], key[14]) = "T 1/4 3/2 4/1");
+    bayesNet.add((keys[3] | keys[9], keys[12]) = "1/4 2/3 3/2 4/1");
+    bayesNet.add((keys[2] | keys[9], keys[12]) = "1/4 8/2 2/3 4/1");
+    bayesNet.add((keys[1] | keys[8], keys[12]) = "4/1 2/3 3/2 1/4");
+    bayesNet.add((keys[0] | keys[8], keys[12]) = "2/3 1/4 3/2 4/1");
 
-  bayesNet.add((key[7] | key[11], key[13]) = "1/4 2/3 3/2 4/1");
-  bayesNet.add((key[6] | key[11], key[13]) = "1/4 3/2 2/3 4/1");
-  bayesNet.add((key[5] | key[10], key[13]) = "4/1 2/3 3/2 1/4");
-  bayesNet.add((key[4] | key[10], key[13]) = "2/3 1/4 3/2 4/1");
+    // Create a BayesTree out of the Bayes net.
+    bayesTree = DiscreteFactorGraph(bayesNet).eliminateMultifrontal();
+  }
+};
 
-  bayesNet.add((key[3] | key[9], key[12]) = "1/4 2/3 3/2 4/1");
-  bayesNet.add((key[2] | key[9], key[12]) = "1/4 8/2 2/3 4/1");
-  bayesNet.add((key[1] | key[8], key[12]) = "4/1 2/3 3/2 1/4");
-  bayesNet.add((key[0] | key[8], key[12]) = "2/3 1/4 3/2 4/1");
+/* ************************************************************************* */
+TEST(DiscreteBayesTree, ThinTree) {
+  const TestFixture self;
+  const auto& keys = self.keys;
 
   if (debug) {
-    GTSAM_PRINT(bayesNet);
-    bayesNet.saveGraph("/tmp/discreteBayesNet.dot");
+    GTSAM_PRINT(self.bayesNet);
```

Comment:
Any reason for not just doing `auto bayesNet = self.bayesNet`? `self` is a little confusing since this is not a class, but this is a minor nitpick.

Reply:
I'm just using the variable from the test fixture...